### DISCUSSION TITLE
[Files service] doubt about how to use POST /bulk with multi-bucket

### DISCUSSION THREAD

**Discussion by theliuk** [mia-platform/community#231] [Q&A] [answered]
### Which section?

Console

### Which is the version?

v2.7.0

### Description

Hi,
I have a doubt about how to use the `POST /bulk` endpoint of `files-service@v2.7.0` with support of scopes. 
From [the docs](https://docs.mia-platform.eu/docs/runtime_suite/files-service/usage#upload-many-post-bulk) I see that you can include the `scope` form data field in your call to upload multiple files. This means that all the files will be under the same scope and there's no way to tell "this file goes to that scope and this other one goes to this scope". 
If there's actually this limit "all files in bulk must go under the same scope" I'd highlight it in the docs.

I call into cause @danibix95 that I believe had something to do with the new feature.

Thanks

**Comment by danibix95** [comment]
@davide-sera

**Reply by seraph-eeno** [reply]
Thank you! :bow:

**Comment by seraph-eeno** [comment, marked as answer]
Hi @theliuk,
what you say is correct. At the moment specifying `scope` in form-data does not produce any effect and all the files are uploaded following the scope taken from the url path of the API call.

As an example, calling the API `POST /pippo-scope/bulk` and providing the body with different files having the `scope` set to `data-form-scope` will upload all the files under the scope `pippo-scope` anyway, ignoring what's written in the body.

It actually seems to me that that example of form-data specified in the docs is a typo.

@fredmaggiowski I could open a PR on the files-service with a better explanation in the docs.

If we want to change the behaviour for the `POST /bulk` and fetch the scope for each file directly from the form-data field, an issue should be opened on the files-service repo directly.

**Reply by theliuk** [reply]
Hi @davide-sera ,
thank you. 

Yeah, I'd update the docs, cause there's no mention of the fact that the scope must be in the path, and I was I was lead to believe that the user had to specify it in the form-data.

Thank you again,

Luca

**Reply by fredmaggiowski** [reply]
I'd update the doc but the fact that the scope form data is accepted when the scope is actually assumed from the path seems like a bug; i'd fix the service to completely ignore the form data and only use the path if that's not necessary

**Reply by seraph-eeno** [reply]
It behaves exactly like this at the moment. Any form-data labeled `scope` is ignored from the Files Service and no business logic is executed on it.
